app/src/main/java/org/oppia/android/app/classroom/classroomlist/ClassroomList.kt

```diff
@@ -0,0 +1,142 @@
+
+/** Test tag for the classroom list. */
+const val CLASSROOM_LIST_TEST_TAG = "TEST_TAG.classroom_list"
+
+/** Displays a list of classroom summaries with a header. */
+@Composable
+fun ClassroomList(
+  classroomSummaryList: List<ClassroomSummaryViewModel>,
+  selectedClassroomId: String
+) {
+  Column(
+    modifier = Modifier
+      .background(
+        color = colorResource(id = R.color.component_color_shared_screen_primary_background_color)
+      )
+      .fillMaxWidth(),
+  ) {
+    Text(
+      text = stringResource(id = R.string.classrooms_list_activity_section_header),
+      color = colorResource(id = R.color.component_color_shared_primary_text_color),
+      fontFamily = FontFamily.SansSerif,
+      fontWeight = FontWeight.Medium,
+      fontSize = 18.sp,
+      modifier = Modifier
+        .testTag(CLASSROOM_HEADER_TEST_TAG)
+        .padding(
+          start = dimensionResource(id = R.dimen.classrooms_text_margin_start),
+          top = dimensionResource(id = R.dimen.classrooms_text_margin_top),
+          end = dimensionResource(id = R.dimen.classrooms_text_margin_end),
+          bottom = dimensionResource(id = R.dimen.classrooms_text_margin_bottom),
+        ),
+    )
+    LazyRow(
+      modifier = Modifier.testTag(CLASSROOM_LIST_TEST_TAG),
+      contentPadding = PaddingValues(
+        start = dimensionResource(id = R.dimen.classrooms_text_margin_start),
+        end = dimensionResource(id = R.dimen.classrooms_text_margin_end),
+      ),
+    ) {
+      items(classroomSummaryList) {
+        ClassroomCard(classroomSummaryViewModel = it, selectedClassroomId)
+      }
+    }
+  }
+}
+
+/** Displays a single classroom card with an image and text, handling click events. */
+@Composable
+fun ClassroomCard(
+  classroomSummaryViewModel: ClassroomSummaryViewModel,
+  selectedClassroomId: String
+) {
+  val screenWidth = LocalConfiguration.current.screenWidthDp
+  val isPortrait = LocalConfiguration.current.orientation == Configuration.ORIENTATION_PORTRAIT
+  val isTablet = if (isPortrait) screenWidth > 600 else screenWidth > 840
+  val isCardSelected = classroomSummaryViewModel.classroomSummary.classroomId == selectedClassroomId
+  Card(
+    modifier = Modifier
+      .height(dimensionResource(id = R.dimen.classrooms_card_height))
+      .width(dimensionResource(id = R.dimen.classrooms_card_width))
+      .padding(
+        start = dimensionResource(R.dimen.promoted_story_card_layout_margin_start),
+        end = dimensionResource(R.dimen.promoted_story_card_layout_margin_end),
+      )
+      .clickable {
+        classroomSummaryViewModel.handleClassroomClick()
+      },
+    backgroundColor = if (isCardSelected) {
+      colorResource(id = R.color.component_color_classroom_card_color)
+    } else {
+      colorResource(id = R.color.component_color_shared_screen_primary_background_color)
+    },
+    border = BorderStroke(2.dp, color = colorResource(id = R.color.color_def_oppia_green)),
+    elevation = 4.dp,
```

Comment:
Please migrate the dp and sp values to dimensions

Reply:
Done.